benchmarks/benchmarks/bench_function_base.py

```diff
@@ -96,46 +96,152 @@ def time_select_larger(self):
-            raise NotImplementedError()
-
-    def time_sort(self, kind, dtype):
-        np.sort(self.e, kind=kind)
-
-    def time_sort_random(self, kind, dtype):
-        np.sort(self.o, kind=kind)
-
-    def time_sort_inplace(self, kind, dtype):
-        self.e.sort(kind=kind)
-
-    def time_sort_equal(self, kind, dtype):
-        self.equal.sort(kind=kind)
-
-    def time_sort_many_equal(self, kind, dtype):
-        self.many_equal.sort(kind=kind)
-
-    def time_argsort(self, kind, dtype):
-        self.e.argsort(kind=kind)
-
-    def time_argsort_random(self, kind, dtype):
-        self.o.argsort(kind=kind)
+    param_names = ['kind', 'dtype', 'array_type']
+
+    # The size of the benchmarked arrays.
+    ARRAY_SIZE = 10000
+    # The size of the unsorted area in the "random unsorted area"
+    # benchmarks
+    AREA_SIZE = 10
+    # ?
+    BUBBLE_SIZE = 10
+
+    def setup(self, kind, dtype, array_type):
+        np.random.seed(1234)
+        self.arr = getattr(self, 'array_' + array_type)(self.ARRAY_SIZE, dtype)
+
+    def time_sort_inplace(self, kind, dtype, array_type):
+        self.arr.sort(kind=kind)
+
+    def time_sort(self, kind, dtype, array_type):
+        np.sort(self.arr, kind=kind)
+
+    def time_argsort(self, kind, dtype, array_type):
+        np.argsort(self.arr, kind=kind)
+
+    def array_random(self, size, dtype):
+        arr = np.arange(size, dtype=dtype)
+        np.random.shuffle(arr)
+        return arr
+    
+    def array_ordered(self, size, dtype):
+        return np.arange(size, dtype=dtype)
+
+    def array_reversed(self, size, dtype):
+        return np.arange(size-1, -1, -1, dtype=dtype)
+
+    def array_uniform(self, size, dtype):
+        return np.ones(size, dtype=dtype)
+
+    def array_sorted_block_10(self, size, dtype):
+        return self.sorted_block(size, dtype, 10)
+
+    def array_sorted_block_100(self, size, dtype):
+        return self.sorted_block(size, dtype, 100)
+
+    def array_sorted_block_1000(self, size, dtype):
+        return self.sorted_block(size, dtype, 1000)
+
+    def array_swapped_pair_1_percent(self, size, dtype):
+        return self.swapped_pair(size, dtype, 0.01)
+
+    def array_swapped_pair_10_percent(self, size, dtype):
+        return self.swapped_pair(size, dtype, 0.1)
+
+    def array_swapped_pair_50_percent(self, size, dtype):
+        return self.swapped_pair(size, dtype, 0.5)
+
+    def array_random_unsorted_area_50_percent(self, size, dtype):
+        return self.random_unsorted_area(size, dtype, 0.5, self.AREA_SIZE)
+
+    def array_random_unsorted_area_10_percent(self, size, dtype):
+        return self.random_unsorted_area(size, dtype, 0.1, self.AREA_SIZE)
+
+    def array_random_unsorted_area_1_percent(self, size, dtype):
+        return self.random_unsorted_area(size, dtype, 0.01, self.AREA_SIZE)
+
+    def array_random_bubble_1_fold(self, size, dtype):
+        return self.random_unsorted_area(size, dtype, 1, size / self.BUBBLE_SIZE)
```

Comment:
@liwt31 I copied the `frac` from your repo, but it seems it's greater than one. Is this purposeful?

Reply:
Yeah It's greater than one in these "bubble" functions and yeah it's a bad variable name. Maybe renaming to `fold` or refactoring the two sets of functions to combine them would be more reasonable?